File: API/src/main/java/com/ltizzi/ecommerce/model/user/UserResponse.java
package com.ltizzi.ecommerce.model.user;

import com.ltizzi.ecommerce.model.cart.CartResponse;
import com.ltizzi.ecommerce.model.purchase.PurchaseResponse;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * @author Leonardo Terlizzi
 */
@Data
@NoArgsConstructor
public class UserResponse {

    private Long user_id;
    private String username;
    private String name;
    private String lastname;
    private String email;
    private String avatar;
    private Date birthday;
    private List<CartResponse> carts = new ArrayList<>();
    private List<PurchaseResponse> purchases = new ArrayList<>();

}

File: Server/src/dto/responses/user.response.ts
import { Review } from "../../models/Review";
import { RoleEnum } from "../../models/RoleEnum";
import { CartResponse } from "./cart.response";
import { ProductResponse } from "./product.response";
import { PurchaseResponse } from "./purchase.response";

export interface UserResponse {
  user_id: number;
  username: string;
  name: string;
  lastname: string;
  email: string;
  avatar: string;
  birthday: Date;
  carts: Array<CartResponse>;
  purchases: Array<PurchaseResponse>;
  roles: Array<RoleEnum>;
  favourites: Array<ProductResponse>;
  reviews: Array<Review>;
}
